docs/data/toolpad/core/components/data-grid/BasicDataProvider.js

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import { createDataProvider } from '@toolpad/core/DataProvider';
+import { DataGrid } from '@toolpad/core/DataGrid';
+import Box from '@mui/material/Box';
+
+const myData = createDataProvider({
```

Comment:
I went over this page and the examples seem using arrays directly seem good, but I wonder if they could be even more clear if they used some kind of mock API with its own methods.
It could be closer to what users will do with real APIs, or maybe it would be more redundant, so not sure, just a thought. I think this works great for now!

The one example that took a bit more for be to understand was the pagination one.

Reply:
I'll think about a solution. In the meantime it should probbaly not be a blocker for this PR